sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -125,32 +139,45 @@ abstract class CodeGenerator[InType <: AnyRef, OutType <: AnyRef] extends Loggin
    * @param objectTerm A possibly boxed version of the result of evaluating this expression.
    */
   protected case class EvaluatedExpression(
-      code: Seq[Tree],
-      nullTerm: TermName,
-      primitiveTerm: TermName,
-      objectTerm: TermName)
+      code: String,
+      nullTerm: String,
+      primitiveTerm: String,
+      objectTerm: String)
+
+  /**
+   * A context for codegen
+   * @param references the expressions that don't support codegen
+   */
+  protected case class CodeGenContext(references: mutable.ArrayBuffer[Expression])
```

Comment:
I think that we can make this into a non-case class with a zero-argument constructor and a single field, `references`, that's initialized to an empty array buffer.